tests/test_cli/test_check_packages.py

```diff
@@ -0,0 +1,284 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Test check packages command module."""
+
+from pathlib import Path
+from typing import Any, List
+from unittest import mock
+
+import pytest
+
+from aea.cli.check_packages import find_all_configuration_files, get_public_id_from_yaml
+from aea.test_tools.test_cases import BaseAEATestCase
+
+
+check_author_patch = mock.patch("aea.cli.check_packages.check_author")
+check_dependencies_patch = mock.patch("aea.cli.check_packages.check_dependencies")
+find_all_packages_ids_patch = mock.patch(
+    "aea.cli.check_packages.find_all_packages_ids", return_value=set()
+)
+
+
+def _unified_yaml_load_patch(**config) -> Any:
+    return mock.patch(
+        "aea.cli.check_packages.unified_yaml_load",
+        return_value=dict(**config),
+    )
+
+
+def _find_all_configuration_files_patch(config_files: List) -> Any:
+    return mock.patch(
+        "aea.cli.check_packages.find_all_configuration_files",
+        return_value=config_files,
+    )
+
+
+class TestCheckPackagesCommand(BaseAEATestCase):
+    """Test check-packages command."""
+
+    use_packages_dir: bool = True
+
+    def test_invocation(
+        self,
+    ) -> None:
+        """Test command invocation."""
+
+        result = self.invoke(
+            "--registry-path",
+            str(self.packages_dir_path),
+            "check-packages",
+        )
+
+        assert result.exit_code == 0, result.output
+        assert all(
+            [
+                str(file) in result.output
+                for file in find_all_configuration_files(
+                    self.packages_dir_path.absolute()
+                )
+            ]
+        )
+
+    def test_dependency_not_found(
+        self,
+    ) -> None:
+        """Test"""
+
+        config_file = (
+            self.t
+            / "packages"
+            / "fetchai"
+            / "agents"
+            / "error_test"
+            / "aea-config.yaml"
+        )
```

Comment:
why not set these in a `setup` method? the gym `connection.yaml` as well. Will make the tests cleaner